src/Microsoft.OData.Core/Microsoft.OData.Core.txt

```diff
@@ -846,7 +846,7 @@ PathParser_TypeCastOnlyAllowedInDerivedTypeConstraint=Type cast segment '{0}' on
 
 ODataResourceSet_MustNotContainBothNextPageLinkAndDeltaLink=A resource set may contain a next page link, a delta link or neither, but must not contain both.
 
-ODataExpandPath_OnlyLastSegmentMustBeNavigationProperty=The last segment, and only the last segment, must be a navigation property in $expand.
+ODataExpandPath_OnlyLastSegmentMustBeNavigationPropertyOrTypeSegment=The last segment, and only the last segment, must be a navigation property or type segment in $expand.

```

Comment:
Does this imply that it's not allowed to have type segments before the last segment?

Reply:
Good catch. 
We can have type segments before the last segments.
The error string should be 
`ODataExpandPath_LastSegmentMustBeNavigationPropertyOrTypeSegment` = `The last segment must be a navigation property or type segment in $expand.`